src/algebra/order/hom/basic.lean

```diff
@@ -43,16 +47,22 @@ class submultiplicative_hom_class (F : Type*) (α β : out_param $ Type*) [has_m
   [has_le β] extends fun_like F α (λ _, β) :=
 (map_mul_le_mul (f : F) : ∀ a b, f (a * b) ≤ f a * f b)
 
-/-- `map_add_le_class F α β` states that `F` is a type of subadditive morphisms. -/
+/-- `mul_le_add_hom_class F α β` states that `F` is a type of subadditive morphisms. -/
 @[to_additive subadditive_hom_class]
 class mul_le_add_hom_class (F : Type*) (α β : out_param $ Type*) [has_mul α] [has_add β] [has_le β]
   extends fun_like F α (λ _, β) :=
 (map_mul_le_add (f : F) : ∀ a b, f (a * b) ≤ f a + f b)
 
+/-- `nonarchimedean_hom_class F α β` states that `F` is a type of non-archimedean morphisms. -/
+class nonarchimedean_hom_class (F : Type*) (α β : out_param $ Type*) [has_add α] [linear_order β]
+  extends fun_like F α (λ _, β) :=
+(map_add_le_max (f : F) : ∀ a b, f (a + b) ≤ max (f a) (f b))
```

Comment:
You could rephrase this without `linear_order` as:
```suggestion
class nonarchimedean_hom_class (F : Type*) (α β : out_param $ Type*) [has_add α] [has_le β]
  extends fun_like F α (λ _, β) :=
(map_add_le_of_left_le_of_right_le (f : F) : ∀ a b c, f a ≤ c → f b ≤ c → f (a + b) ≤ c)
```

Reply:
I would really prefer to keep the `map_add_le_max` version. The most immediate application I have for this is to define nonarchimedan (semi)norms on additive groups and rings (and the corresponding hom classes); for this, `β` will be the real numbers. I also expect this class to be used with `β = nnreal` or, more generally, a `linear_ordered_comm_monoid_with_zero`, but I never plan to use it without a linear order on `β`.

Reply:
Okay, then this looks good to me. Thanks for the explanation!